File: threedollar-domain/src/main/java/com/depromeet/threedollar/domain/boss/domain/account/repository/BossAccountRepositoryCustom.java
package com.depromeet.threedollar.domain.boss.domain.account.repository;

import com.depromeet.threedollar.domain.boss.domain.account.BossAccount;
import com.depromeet.threedollar.domain.boss.domain.account.BossAccountSocialType;

public interface BossAccountRepositoryCustom {

    BossAccount findBossAccountBySocialTypeAndSocialId(String socialId, BossAccountSocialType socialType);

    BossAccount findBossAccountById(Long bossAccountId);

}

File: threedollar-domain-mongo/src/main/kotlin/com/depromeet/threedollar/domain/mongo/boss/domain/account/repository/BossAccountRepositoryCustom.kt
package com.depromeet.threedollar.domain.mongo.boss.domain.account.repository

import com.depromeet.threedollar.domain.mongo.boss.domain.account.BossAccount
import com.depromeet.threedollar.domain.mongo.boss.domain.account.BossAccountSocialType

interface BossAccountRepositoryCustom {

    fun existsBossAccountById(id: String): Boolean

    fun findBossAccountById(id: String): BossAccount?

    fun findBossAccountBySocialInfo(socialId: String, socialType: BossAccountSocialType): BossAccount?

    fun existsBossAccountBySocialInfo(socialId: String, socialType: BossAccountSocialType): Boolean

}
